src/main/java/fi/aalto/cs/apluscourses/utils/Callbacks.java

```diff
@@ -0,0 +1,72 @@
+package fi.aalto.cs.apluscourses.utils;
+
+import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.model.Module;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class Callbacks {
+
+  @NotNull
+  private final List<Method> postDownloadModuleCallbacks = new ArrayList<>();
+
+  private static void addCallbacksFromArray(@Nullable JSONArray callbackArray,
+                                            @NotNull String methodName,
+                                            @NotNull List<Method> callbackList) throws ClassNotFoundException {
+    if (callbackArray == null) {
+      return;
+    }
+
+    for (int i = 0; i < callbackArray.length(); i++) {
+      String className = callbackArray.getString(i);
+
+      Class<?> callbackClass = Class.forName("fi.aalto.cs.apluscourses.utils.callbacks." + className);
+      Method method = Arrays.stream(callbackClass.getDeclaredMethods())
+          .filter(m -> m.getName().equals(methodName))
+          .findFirst()
+          .orElseThrow();
```

Comment:
I don't think we should use reflection here (even though this looks reasonably secure).

I suggest we just use a single dictionary of callbacks (`Map<String, Callback>`) where all the available callbacks are registered.

Reply:
For once I agree that reflection here isn't the best idea. No type safety and questionable extensibility since adding new callbacks requires updating the plugin anyway.
Reflection was my first thought when implementing this, but in hindsight it's not a good solution here :(((